src/backend/distributed/planner/merge_planner.c

```diff
@@ -243,14 +243,27 @@ CreateNonPushableMergePlan(Oid targetRelationId, uint64 planId, Query *originalQ
 
 	CitusTableCacheEntry *targetRelation = GetCitusTableCacheEntry(targetRelationId);
 
-	/*
-	 * Get the index of the column in the source query that will be utilized
-	 * to repartition the source rows, ensuring colocation with the target
-	 */
-	distributedPlan->sourceResultRepartitionColumnIndex =
-		SourceResultPartitionColumnIndex(mergeQuery,
-										 sourceQuery->targetList,
-										 targetRelation);
+
+	if (IsCitusTableType(targetRelation->relationId, SINGLE_SHARD_DISTRIBUTED))
+	{
+		/*
+		 * if target table is SINGLE_SHARD_DISTRIBUTED let's set this to invalid -1
+		 * so later in execution phase we don't rely on this value and try to find single shard of target instead.
```

Comment:
is this an existing design for  SINGLE_SHARD_DISTRIBUTED

Reply:
Yes @tejeswarm , while execution we call `ShardIdForTuple` function , there for single sharded and local table this value should be -1